robot/client/client.go

```diff
@@ -288,6 +288,35 @@ func New(ctx context.Context, address string, clientLogger logging.ZapCompatible
 		return nil, err
 	}
 
+	// If running in a testing environment, wait for machine to report a state of
+	// running. We often establish connections in tests and expected resources to
+	// be immediately available once the web service has started; resources will
+	// not be available when the machine is still initializing.
+	//
+	// It is expected that golang SDK users will handle lack of resource
+	// availability due to the machine being in an initializing state themselves.
+	if testing.Testing() {
```

Comment:
can we add this block inside robottestutils.NewRobotClient instead?

Reply:
can we make change all `client.New`s to `robottestutils.NewRobotClient` a ticket? I think it would be good to segregate the testing code from actual code if possible.

Reply:
For sure: https://viam.atlassian.net/browse/RSDK-9609.